subscribers.js

```diff
@@ -1,7 +1,95 @@
 "use strict";
 
-// Set of all registered emails.
-// TODO: Implement a real persistent storage solution.
-const gEmails = new Set();
+const pg = require("pg");
 
-module.exports = gEmails;
+const AppConstants = require("./app-constants");
+
+const dbConfig = {
+  database: AppConstants.DATABASE_NAME,
+  host: AppConstants.DATABASE_HOST,
+  port: AppConstants.DATABASE_PORT,
+  max: 10,
+};
+
+const dbPool = new pg.Pool(dbConfig);
+
+function dbq(aQuery) {
+  return dbPool.query(aQuery);
+}
+
+function _ret(aSuccess, aError, aOther) {
+  return Object.assign({
+    success: aSuccess,
+    error: aError || null,
+  }, aOther);
+}
+
+const Subscribers = {
+  async addUser(aEmail) {
+    // Error code for attempting to add a duplicate entry on a UNIQUE key.
+    const DUPLICATE_ERROR = 23505;
+    try {
+      await dbq({
+        text: "INSERT INTO users ( email ) VALUES ( $1 );",
+        values: [ aEmail ],
+      });
+
+      return _ret(true);
+    } catch (e) {
+      if (e.code == DUPLICATE_ERROR) {
+        // Duplicate entry, count it as success.
+        return _ret(true, null, { duplicate: true });
+      }
+
+      return _ret(false, e);
+    }
+  },
+
+  async deleteUser(aEmail) {
+    try {
+      await dbq({
+        text: "DELETE FROM users WHERE email = $1;",
+        values: [ aEmail ],
+      });
+      return _ret(true);
+    } catch(e) {
+      return _ret(false, e);
+    }
+  },
+
+  async getUser(aEmail) {
+    try {
+      const rows = (await dbq({
+        text: "SELECT * FROM users WHERE email = $1;",
+        values: [ aEmail ],
+      })).rows;
+
+      if (!rows.length) {
+        return _ret(false);
```

Comment:
This feels like it should be more verbose on why it was an error.
Perhaps something like:

```js
return _ret(false, "Entry was not found");
// or
return _ret(false, new Error("Entry was not found"));
```